airbyte-integrations/base-singer/base_singer/singer_helpers.py

```diff
@@ -104,14 +104,15 @@ def create_singer_catalog_with_selection(masked_airbyte_catalog, discovered_sing
         for singer_stream in discovered_singer_catalog.get("streams"):
             if singer_stream.get("stream") in stream_to_airbyte_schema:
                 new_metadatas = []
-                metadatas = singer_stream.get("metadata")
-                for metadata in metadatas:
-                    new_metadata = metadata
-                    new_metadata["metadata"]["selected"] = True
-                    if not is_field_metadata(new_metadata):
-                        new_metadata["metadata"]["forced-replication-method"] = "FULL_TABLE"
-                    new_metadatas += [new_metadata]
-                singer_stream["metadata"] = new_metadatas
+                if singer_stream.get("metadata"):
```

Comment:
metadata can be null (e.g. in the case for exchangeratesapi). we need unit tests for this.

Reply:
yeah I'm working on tests